### PR TITLE
ES sort functionality supported with configurable sort options

### PR DESCRIPTION
**Pull request by jdiouf**

### CODE REVIEW

File: app/core/migrations/0002_searchsortoption.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        ('core', '0001_initial'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='SearchSortOption',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('display_name', models.CharField(help_text='Enter the display name of the sorting option', max_length=200)),
+                ('field_name', models.CharField(help_text='Enter the metadata field name as displayed in Elasticsearch             e.g. course.title', max_length=200)),
```
Comment (mattdrayer) [inline app/core/migrations/0002_searchsortoption.py:19]: Nit: looks like we have some odd spacing in the value for this field.

File: app/core/models.py [hunk 2]
```diff
@@ -80,3 +80,28 @@ def get_absolute_url(self):
     def __str__(self):
         """String for representing the Model object."""
         return f'{self.id}'
+
+
+class SearchSortOption(models.Model):
```
Comment (mattdrayer) [inline app/core/models.py:85]: Recommend inheriting from TimeStampedModel for all of our Django models in order to add some automatic created/modified date stamping capability:  https://django-model-utils.readthedocs.io/en/latest/models.html

File: app/es_api/utils/queries.py [hunk 3]
```diff
@@ -43,7 +43,7 @@ def add_search_filters(search, filters):
     result_search = search
 
     for filter_name in filters:
-        if filter_name != 'page':
+        if filter_name != 'page' and filter_name != 'sort':
             # logger.info(filter_name)
```
Comment (mattdrayer) [inline app/es_api/utils/queries.py:47]: Remove disabled debug code before merging to main branch.

File: app/es_api/utils/queries.py [hunk 4]
```diff
@@ -54,6 +54,28 @@ def add_search_filters(search, filters):
...
+    """This helper method checks if one of the configured sort options was
+        sent with the request then adds it to the search query
+        search = Elasticsearch Search object
+        filters = object containing all the request parameters
+        returns -> modified Elasticsearch search object"""
+    # add sort if it was sent in the request
+    result_search = search
+
+    if 'sort' in filters:
+        key = filters['sort']
```
Comment (mattdrayer) [inline app/es_api/utils/queries.py:67]: Is there any sort of input validation that we need to be concerned with here?

File: app/es_api/utils/queries.py [hunk 5]
```diff
@@ -83,7 +108,8 @@ def search_by_keyword(keyword="", filters={}):
 
     # call to elasticsearch to execute the query
     response = s.execute()
-    logger.info(response)
+    # logger.info(response)
```
Comment (mattdrayer) [inline app/es_api/utils/queries.py:111]: Remove debug code before merging to main branch

File: app/es_api/views.py [hunk 6]
```diff
@@ -26,6 +26,9 @@ def search_index(request):
     if (request.GET.get('p')) and (request.GET.get('p') != ''):
         filters['page'] = request.GET['p']
 
+    if (request.GET.get('sort')) and (request.GET.get('sort') != ''):
+        filters['sort'] = request.GET['sort']
+
     if keyword != '':
         errorMsg = {
             "message": "error executing ElasticSearch query; " +
```
Comment (mattdrayer) [inline app/es_api/views.py:34]: We should include an action for the user to take in this situation.

**Review by mattdrayer — APPROVED** [review]
Looks good!  Django makes this "sort" of thing so easy 😄

File: app/es_api/utils/queries.py (hunk 4 above)
Comment (jdiouf) [reply app/es_api/utils/queries.py:67]: On line 68-72, we basically check that whatever field was sent over matches one of the configured keywords in the database. If not then we just ignore it.

Comment (mattdrayer) [reply app/es_api/utils/queries.py:67]: Makes sense thanks

**Review by KarenAJ — APPROVED** [review]
Approved!

**Pull request merged**